src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
+
+        $qb = $this->em->getRepository(AuditLog::class)->createQueryBuilder('al')
+            ->join('al.source', 's')
+            ->where('s.class = :attrClass')->setParameter('attrClass', $attrClass)
+            ->andWhere("al.action = 'insert'")
+            ->andWhere("al.diff LIKE :diffLikeClsS ESCAPE '°' OR al.diff LIKE :diffLikeClsN ESCAPE '°'")
+            ->setParameter('diffLikeClsS', '%"'.$entInAttr.'":%"class":'.$entClassJson.'%,"fk":"'.$entId.'",%') // string fk
+            ->setParameter('diffLikeClsN', '%"'.$entInAttr.'":%"class":'.$entClassJson.'%,"fk":'.$entId.',%') // numeric fk
+        ;
+        $i = 1;
+        foreach ($additionalConditions as $condition) {
+            $condValue = json_encode($condition['value']);
+            $qb->andWhere('al.diff LIKE :diffLike'.$i)->setParameter('diffLike'.$i, '%"'.$condition['attr'].'"%"new":'.$condValue.'%');
+            ++$i;
+        }
+        $candidates = $qb->getQuery()->getResult();
+        /// checking real values
+        $matchingIds = array();
+        /** @var AuditLog $candidate */
+        foreach ($candidates as $candidate) {
+            $diff = $candidate->getDiff();
+            if ($diff[$entInAttr]['new']['fk'] != $entId) {
+                continue; // wrong id => next candidate
+            }
+            foreach ($additionalConditions as $condition) {
+                if ($condition['value'] !== $diff[$condition['attr']]['new']) {
+                    continue 2; // condition not fulfilled => next candidate
+                }
+            }
+            $matchingIds[] = $candidate->getSource()->getFk();
+        }
+
+        return array('class' => $attrClass, 'ids' => $matchingIds);
+    }
+
+    /**
+     * Extends the QueryBuilder from getAllVersionsQb with logs of attributes the entity is not the owning side.
+     *
+     * @param QueryBuilder $qb
+     * @param string       $attributeClass probably ->getInverseSideAttributeIds()['class']
+     * @param int[]        $ids            probably ->getInverseSideAttributeIds()['ids']
+     */
+    protected function extendQbWithInverseSideAttribute($qb, $attributeClass, $ids)
+    {
+        $unique = count($qb->getParameters()).substr($attributeClass, (strrpos($attributeClass, '\\') ?: -1) + 1);
+        $qb->orWhere('s.fk IN (:idsAttr'.$unique.') AND s.class = :classAttr'.$unique)
+            ->setParameter('classAttr'.$unique, $attributeClass)
+            ->setParameter('idsAttr'.$unique, $ids)
+        ;
+    }
+
+    /**
+     * Get the label for a CustomField.
+     *
+     * To be called in getColumnNameForAssociation() for instances using
+     * CubeTools\CubeCustomFieldsBundle\Entity\EntityCustomField
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string
+     */
+    protected function getCustomFieldLabel(AuditLog $currentVersion)
+    {
+        if (!$currentVersion->getTarget() || 'custom_fields' !== $currentVersion->getTarget()->getTbl()) {
+            return;
```

Comment:
You declare in phpdoc that you return string, but here you return void. Although, as written before, I am not a fan of return inside body (would rather make all in if else statement and at the end return value). You could update phpdoc with 'string|void' or return here false and also update phpdoc.

Reply:
correct